### ISSUE TITLE
Let's remove page objects from @bigtest/interaction...

### ISSUE THREAD

**Issue by wwilsman** [thefrontside/bigtest#81]
`@bigtest/interaction` is named after the `Interaction` class, but it's primary usage is with page objects. The page objects themselves are just thin wrappers around interactions. They only add property getters that may throw on certain conditions, otherwise their methods just get forwarded to the interaction. The term "page object" is actually a misnomer within our library because we can make much smaller, composable, component-based, interaction objects... in fact we should just call them that: _**Interaction Objects.**_ 

Although, do we need the thin layer objects to begin with? `@interaction` (formerly `@page`) already  builds custom `Interaction` subclasses. These custom subclasses are returned when page-object property methods are called. We would just need to move the other getter properties down to the custom interaction subclasses, and page-objects can be removed completely.

If we do add the property getters to interactions, that adds a potential pitfall:
```javascript
new CustomInteraction()
  .fillName('name')
  .click('.submit')
  .someProperty 
// ^ this is just a getter that returns a value;
// the above interactions never actually happen.
```

However, this actually helps when composing interactions, and can be converged on using the interaction's own convergence methods
```javascript
// when composing interactions ...
let interaction = new CustomInteraction()
  .fillName('name')
  .click('.submit')

// ... this works ...
expect(interaction.someProperty).to.equal('something')

// ... and you can converge on values via convergence methods
let afterProp = await interaction
  .do(() => interaction.someProperty)

// or write the assertions within the interaction convergence
new CustomInteraction()
  .fillName('name')
  .click('.submit')
  .once(function() {
    expect(this.someProperty).to.equal('something')
  })
```

#### What does this mean for existing page objects?
Well, _nothing_. Page objects are such a thin wrapper that by just moving the getters down to the underlying interaction, the API stays exactly the same. We would just alias `@page` to `@interaction`, deprecate `@page`, and remove it for the v1 release.

_Technically_, any use of `page.interaction` would need to be replaced with just `page` since the interaction would no longer be wrapped. But that usage is pretty minimal.

**Comment by cowboyd** [comment]
I agree 100% that PageObject is a complete misnomer.

Although I'm still confused on terminology. Who is responsible for scoping to the DOM? Because to me, the word "interaction" feels both atomic and abstract. Like a click, or touch, or something composed of touches like a swipe. 

Although when I actually read it, I like the sound of 

```js
@interaction class CardStackInteraction {
  cards = collection(class CardInteraction {
    title = focusable('input[name="title"]');
    handle = draggable('.handle');
  })
}
```

Emphasis on _`@interaction class`_

Total  aside: Do we currently making the base interaction class a `find` convergence on the element, and then all of its interaction properties just chain off the parent?

**Comment by wwilsman** [comment]
If we change the naming of the interaction class to `Interactor`, the term _interactions_ could refer to the importable interaction helpers: `fillable`, `clickable`, `focusable`, `etc-able`. And the responsibility of scoping to the DOM would be the responsibility of the _Interactor_.

```javascript
@interactor class CardStackInteractor {
  cards = collection('.card', class CardInteractor {
    title = focusable('input[name="title"]');
    handle = draggable('.handle');
  });
}

let cardStack = new CardStackInteractor('.card-stack')
```

_Aside:_ That's not how it is currently done, but could very well be. Currently, the scope is given to the query selector function: `$('.selector', this.$scope)`. `this.$scope` is a getter that will throw when then scope does not exist. `$()` will throw if `'.selector'` does not exist. So in that single convergence, it will converge on both the scope and element existing.

**Comment by cowboyd** [comment]
Yeah, although I'm gonna play devil's advocate against my own devilish advocation. If we do think of interactions as a tree of composed convergences, then the separation between interaction and interactor is a false one.

I'm not sure of anything anymore :)
